src/main/java/ui/components/pickers/MilestonePickerDialog.java

```diff
@@ -0,0 +1,308 @@
+package ui.components.pickers;
+
+import backend.resource.TurboIssue;
+import backend.resource.TurboMilestone;
+import javafx.application.Platform;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.*;
+import javafx.scene.layout.FlowPane;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+import javafx.util.Pair;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+public class MilestonePickerDialog extends Dialog<Pair<ButtonType, Integer>> {
+    private static final String OCTICON_ARROW = "\uf03e";
+    private static final String DIALOG_TITLE = "Select Milestone";
+
+    private final List<PickerMilestone> originalMilestones = new ArrayList<>();
+    private FlowPane assignedMilestonePane;
+    private VBox matchingMilestonesPane;
+    private TextField inputField;
+    private MilestonePickerState state;
+
+    /**
+     * The issue and the originalMilestones list provided should come from the same repository
```

Comment:
I think doing that will unnecessarily increase the coupling. MilestonePickerDialog should not need to know of the existing models.